duolingo.py

```diff
@@ -28,13 +29,21 @@ class DuolingoException(Exception):
 class Duolingo(object):
     USER_AGENT = "Python Duolingo API/{}".format(__version__)
 
-    def __init__(self, username, password):
+    def __init__(self, username, password, *, jwt=None, session_file=None):
+        """
+        :param username: Username to use for duolingo
+        :param password: Password to authenticate as user. Required, in order to access detailed user data.
+        :param jwt: Duolingo login token. Will be checked and used if it is valid.
+        :param session_file: File path to a file that the session token can be stored in, to save repeated login
+        requests. Username and password are still required when using session file, as token may expire.
+        """
         self.username = username
         self.password = password
+        self.session_file = session_file
         self.user_url = "https://duolingo.com/users/%s" % self.username
         self.session = requests.Session()
         self.leader_data = None
-        self.jwt = None
+        self.jwt = jwt
 
         if password:
```

Comment:
Is it still required here? Since we can login now without password.

Reply:
Hmm, well we probably should remove this if statement, simply because password is required, and we alway want to do a _login() to check the provided jwt is valid? But I think that's the opposite of what you were suggesting?

Reply:
Well, we should remove this statement anyway :) But I don't think password has to be a required argument.

Reply:
Okay, changed it up in the new changes, to not require password, as long as jwt or session_file are specified